lib/internal/Magento/Framework/Api/Test/Unit/Api/SimpleDataObjectConverterTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\Api\Test\Unit\Api;
+
+use Magento\Framework\Api\SimpleDataObjectConverter;
+
+/**
+ * Unit test class for \Magento\Framework\Api\SimpleDataObjectConverter
+ */
+class SimpleDataObjectConverterTest extends \PHPUnit\Framework\TestCase
+{
+    public function testCamelCaseToSnakeCase()
+    {
+        $this->assertEquals("default_shipping", SimpleDataObjectConverter::camelCaseToSnakeCase("defaultShipping"));
+    }
+
+    public function testCamelCaseToSnakeCaseWithNumber()
+    {
+        $this->assertEquals("default_shipping_1", SimpleDataObjectConverter::camelCaseToSnakeCase("defaultShipping1"));
+    }
+
+    public function testSnakeCaseToCamelCase()
+    {
+        $this->assertEquals("defaultShipping", SimpleDataObjectConverter::snakeCaseToCamelCase("default_shipping"));
+    }
+
+    public function testSnakeCaseToCamelCaseWithNumber()
+    {
+        $this->assertEquals("defaultShipping1", SimpleDataObjectConverter::snakeCaseToCamelCase("default_shipping_1"));
```

Comment:
Could you add extra test case - for example `defaultShipping123` ?

Reply:
Thanks @lbajsarowicz for pointing out the issue. I have updated the regex and add a few more test cases.

Though I still worried about a thing!
Earlier:
 `defaultShipping1` -> `default_shipping1`
Now:
 `defaultShipping1` -> `default_shipping_1`

If my column is `default_shipping1` then it will not store data now. I think this may break the 3rd party modules?